cartographer/mapping/internal/2d/overlapping_submaps_trimmer.h

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2018 The Cartographer Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CARTOGRAPHER_MAPPING_INTERNAL_2D_OVERLAPPING_SUBMAPS_TRIMMER_H_
+#define CARTOGRAPHER_MAPPING_INTERNAL_2D_OVERLAPPING_SUBMAPS_TRIMMER_H_
+
+#include "cartographer/common/port.h"
+#include "cartographer/mapping/pose_graph_trimmer.h"
+
+namespace cartographer {
+namespace mapping {
+
+// Trims submaps that have less than 'min_covered_cells_count' cells not
+// overlapped by at least 'fresh_submaps_count` submaps.
+class OverlappingSubmapsTrimmer : public PoseGraphTrimmer {
+ public:
+  OverlappingSubmapsTrimmer(uint16 fresh_submaps_count,
+                            uint16 min_covered_cells_count)
+      : fresh_submaps_count_(fresh_submaps_count),
+        min_covered_cells_count_(min_covered_cells_count) {}
+  ~OverlappingSubmapsTrimmer() override = default;
+
+  void Trim(Trimmable* pose_graph) override;
```

Comment:
This will only work in 2D. (The documentation for this is that the header is in the 2d folder.)
Does this fail early when I pass a 3D pose graph?
Or do you want to rename the class to include 2D?

Reply:
renamed the class, moved the files.